cli/cmd/profile.go

```diff
@@ -87,3 +126,129 @@ func renderProfileTemplate(config *templateConfig, w io.Writer) error {
 	_, err = w.Write(buf.Bytes())
 	return err
 }
+
+func renderOpenAPI(options *profileOptions, w io.Writer) error {
+	var input io.Reader
+	if options.openAPI == "-" {
+		input = os.Stdin
+	} else {
+		var err error
+		input, err = os.Open(options.openAPI)
+		if err != nil {
+			return err
+		}
+	}
+
+	bytes, err := ioutil.ReadAll(input)
+	if err != nil {
+		return fmt.Errorf("Error reading file: %s", err)
+	}
+	json, err := yaml.YAMLToJSON(bytes)
+	if err != nil {
+		return fmt.Errorf("Error parsing yaml: %s", err)
+	}
+
+	swagger := spec.Swagger{}
+	err = swagger.UnmarshalJSON(json)
+	if err != nil {
+		return fmt.Errorf("Error parsing OpenAPI spec: %s", err)
+	}
+
+	profile := sp.ServiceProfile{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s.%s.svc.cluster.local", options.name, options.namespace),
+			Namespace: controlPlaneNamespace,
+		},
+		TypeMeta: meta_v1.TypeMeta{
+			APIVersion: "linkerd.io/v1alpha1",
+			Kind:       "ServiceProfile",
+		},
+	}
+
+	routes := make([]*sp.RouteSpec, 0)
+	for path, item := range swagger.Paths.Paths {
+		pathRegex := pathToRegex(path)
+		if item.Delete != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodDelete, item.Delete.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Get != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodGet, item.Get.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Head != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodHead, item.Head.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Options != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodOptions, item.Options.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Patch != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPatch, item.Patch.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Post != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPost, item.Post.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Put != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPut, item.Put.Responses)
+			routes = append(routes, spec)
+		}
+	}
+
+	profile.Spec.Routes = routes
```

Comment:
i think it's useful to make the output of `profile --open-api` commands stable. the `swagger.Paths.Paths` map above will be intentionally randomized by Go. you can stabilize it here:
```go
sort.Slice(routes, func(i, j int) bool {
	return routes[i].Name < routes[j].Name
})
```

Reply:
I went with sorting the path keys prior to constructing routes